src/DonationForms/Blocks/DonationFormBlock/resources/app/Components/ModalForm.tsx

```diff
@@ -1,93 +1,43 @@
-import {useEffect, useRef, useState} from '@wordpress/element';
+import {useState} from '@wordpress/element';
 import IframeResizer from 'iframe-resizer-react';
-import {createPortal} from 'react-dom';
 
 import '../../editor/styles/index.scss';
-import {useCallback} from 'react';
+import FormModal from '../../common/FormModal';
 
 type ModalFormProps = {
     dataSrc: string;
     embedId: string;
     openFormButton: string;
-    openByDefault?: boolean;
     isFormRedirect: boolean;
     formViewUrl: string;
 };
 
 /**
+ * @unreleased
  * @since 3.4.0
  * @since 3.2.0 include types. update BEM classnames.
  * @since 3.0.0
  */
-export default function ModalForm({dataSrc, embedId, openFormButton, openByDefault, isFormRedirect, formViewUrl}: ModalFormProps) {
-    const [isOpen, setIsOpen] = useState(openByDefault || isFormRedirect);
-    const modalRef = useRef(null);
+export default function ModalForm({dataSrc, embedId, openFormButton, isFormRedirect, formViewUrl}: ModalFormProps) {
     const [dataSrcUrl, setDataSrcUrl] = useState(dataSrc);
 
     const resetDataSrcUrl = () => {
-         if (!isOpen && isFormRedirect) {
+        if (isFormRedirect) {
             setDataSrcUrl(formViewUrl);
         }
     };
 
-    const toggleModal = () => {
-        setIsOpen(!isOpen);
-
-        resetDataSrcUrl();
-    };
-
-    useEffect(() => {
-        const {current: el} = modalRef;
-        if (isOpen) {
-            el.showModal();
-        }
-    }, [isOpen]);
-
```

Comment:
Removing this logic is going to be problematic for offline gateways like Stripe that refresh the page and need to programmatically open the confirmation page from the modal.  I could have done a better job explaining that here using comments as it was adding recently to fix that very issue.

Reply:
Example:

<img width="930" alt="Screenshot 2024-03-13 at 11 53 21 AM" src="https://github.com/impress-org/givewp/assets/10138447/d6264f80-55ec-4745-9e59-0b02e99e8d17">

Reply:
Yea I took a look at your PR for it the other night when it caught my eye. I thought it would still work as I was passing resetDataSrcUrl to onOpenChange. I should have tested it. In any case we can control the modal. 

It looks like it was limited to the frontend previously. fe95f70572e2a897b3d227fc097d21a04d42c6e2

Reply:
Also how's this? af6a9bfb9a95430d52aa438917bb7f1d1d471c85

Reply:
great thanks for the update!